craft_application/models/project.py

```diff
@@ -131,13 +130,14 @@ def _vectorise_architectures(cls, values: str | list[str]) -> list[str]:
     @pydantic.field_validator("build_on", "build_for", mode="after")
     @classmethod
     def _validate_architectures(cls, values: list[str]) -> list[str]:
-        """Validate the architecture entries."""
-        for architecture in values:
-            if architecture != "all" and not is_valid_architecture(architecture):
-                raise errors.CraftValidationError(
-                    f"Invalid architecture: {architecture!r} "
-                    "must be a valid debian architecture."
-                )
+        """Validate the architecture entries.
+
+        Entries must be a valid debian architecture or 'all'. Architectures
+        may have an optional base prefix formatted as '[<base>:]<arch>'.
+
+        :raises ValueError: If any of the architectures are not valid.
```

Comment:
And then I presume this wouldn't be restricted only to architectures, invalid bases could also result in `ValueError` being raised (so maybe "bases or architectures are not valid"?)

Reply:
Reworded